apollo-federation/src/link/cost_spec_definition.rs

```diff
@@ -119,35 +120,36 @@ impl CostSpecDefinition {
     }
 
     pub(crate) fn cost_directive(
-        &self,
         schema: &FederationSchema,
         arguments: Vec<Node<Argument>>,
     ) -> Result<Directive, FederationError> {
-        let name = self
-            .directive_name_in_schema(schema, &COST_DIRECTIVE_NAME_IN_SPEC)?
-            .unwrap_or(COST_DIRECTIVE_NAME_DEFAULT);
+        let name =
+            Self::cost_directive_name(schema)?.ok_or_else(|| SingleFederationError::Internal {
+                message: "The \"@cost\" directive is undefined in the target schema".to_string(),
+            })?;
```

Comment:
we have a convenient macro for this:

```suggestion
            Self::cost_directive_name(schema)?.ok_or_else(|| 
                internal_error!("The \"@cost\" directive is undefined in the target schema")
            )?;
```